### DISCUSSION TITLE
Migrations failed: When trying to install apt and pypi packages (both) in airflow

### DISCUSSION THREAD

**Discussion by adsi7698** [apache/airflow#22977] [Q&A]
**My dockerfile -**
```
FROM apache/airflow:2.2.3

COPY /airflow/requirements.txt /requirements.txt

RUN pip install --no-cache-dir -r /requirements.txt

USER root

# Update aptitude with new repo
RUN apt-get update

# Install software 
RUN apt-get install -y git

USER airflow
```

reference for this dockerfile - https://airflow.apache.org/docs/docker-stack/build.html#a-pypi-package-example

While I install my helm chart for airflow, it fails at migration for webserver, worker, scheduler and sometimes triggerer as well. All fails with init migrations container.

Error in migrations file -
```
File "/home/airflow/.local/lib/python3.7/site-packages/alembic/script/base.py", line 259, in _catch_revision_errors
raise util.CommandError(resolution) from re
alembic.util.exc.CommandError: Can't locate revision identified by 'SomeNumberHere'
```

My question is - 
Is my dockerfile right?
What are the basic steps to debug this error? (k8s, openshift, docker, airflow)

**Comment by potiuk** [comment]
It has nothing to do with dockerfile. The problem is that your database is somewhat wrongly initialized - here is the list of migrations https://airflow.apache.org/docs/apache-airflow/stable/migrations-ref.html and apparently your alembic "version" table got somewhat broken or stores a migration that is not present in the official list. It could happen if your db was tampered with or used pre-release version or something like that. My recommendation - wipe your db and start from scratch if none of it makes sense for you.

**Reply by adsi7698** [reply]
Thank you for replying :)

alembic.util.exc.CommandError: Can't locate revision identified by '587bdf053233'

I can see this reference in the link you provided. 

By deleting database, if I am not wrong means to delete PVC?

**Reply by potiuk** [reply]
No idea what you mean by that. I think I cannot advice you what exactly to delete so that I am not responsible for any harm. 
But I think you have bigger problems with your - but what they are - hard to say. I tihnk you need to understand what went wrong, which data to delete (and consequences of it) yourself. You just need to know what you are doing. 

If I tell you something - will you sue me if I advised wrongly? Or will you tell your boss "this guy told me and I did it"? 

Make all necessary backups, experiment, do a test run on staging sytem and see what you come up with.

**Reply by adsi7698** [reply]
Ohh nono, I am just testing it out. It won't matter even if anything gets deleted.

I already deleted persistent volumes and re installed everything, still it says the same error.

**Reply by potiuk** [reply]
Still no idea where it could come from - it's strange error you get. Never saw it.
